packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/MultiFrameViewer/MultiFrameViewerContainer.spec.js

```diff
@@ -140,7 +140,30 @@ describe('Component > MultiFrameViewerContainer', function () {
         expect(image).to.have.lengthOf(1)
         expect(image.prop('xlinkHref')).to.equal('https://some.domain/image.jpg')
       })
-    })  
+    })
+
+    describe('with marks from an active drawing or transcription task tool', function () {
+      const validateSpy = sinon.spy()
+
+      // testMarks is a generic Map, does not represent actual marks from a task tool
+      const testMarks = new Map([
+        [1, 'one'],
+        [2, 'two'],
+        [3, 'three']
+      ])
+
+      // mock an active drawing or transcription task tool:
+      const activeTool = {
+        marks: testMarks,
+        validate: validateSpy
+      }
+
+      it('should validate active tool marks on frame change', function () {
+        wrapper.setProps({ activeTool })
+        wrapper.instance().onFrameChange(2)
+        expect(validateSpy).to.have.been.calledOnce()
+      })
```

Comment:
Not a blocking comment, but `wrapper.instance()` should be avoided in tests, since it's null for functional components.

This test doesn't tell us what validation does. If the expectation is that incomplete marks are deleted, then a better test would set up an incomplete translation line and expect it to be deleted on frame change. That would be a behaviour-driven test, whereas this tests how the code is implemented. Implementation can change, in refactors, without changing behaviour, which makes tests like this one brittle.

Reply:
Yes I agree, I'll refactor this test better asap.

Reply:
No hurry to do this right now, by the way.

Reply:
@eatyourgreens I think this test is better now, not ideal (still using `wrapper.instance()`), but since #2808 refactors to functional component and relatedly refactors tests with mount then we can include removing `wrapper.instance()` in that PR? Sorry that #2808 will need to be refactored because of this PR, but I can help or will prioritize review.

Reply:
Sounds good. I wouldn’t worry about breaking #2808. 

I think the equivalent fix, using hooks, would be something like
```js
useEffect(() => {
 activeTool?.validate()
}, [frame])
```

Reply:
New test looks good. 👍🏻